pkg/apis/kubevirt/validation/worker.go

```diff
@@ -19,38 +19,61 @@ import (
 
 	apiskubevirt "github.com/gardener/gardener-extension-provider-kubevirt/pkg/apis/kubevirt"
 
-	corev1 "k8s.io/api/core/v1"
+	gardencorev1 "github.com/gardener/gardener/pkg/apis/core"
+	k8scorev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 )
 
 // ValidateWorkerConfig validates a WorkerConfig object.
-func ValidateWorkerConfig(config *apiskubevirt.WorkerConfig, fldPath *field.Path) field.ErrorList {
+func ValidateWorkerConfig(config *apiskubevirt.WorkerConfig, dataVolumes []gardencorev1.DataVolume, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
 	if config.DNSPolicy != "" {
 		dnsPolicyPath := fldPath.Child("dnsPolicy")
 		dnsConfigPath := fldPath.Child("dnsConfig")
 
 		switch config.DNSPolicy {
-		case corev1.DNSDefault, corev1.DNSClusterFirstWithHostNet, corev1.DNSClusterFirst, corev1.DNSNone:
+		case k8scorev1.DNSDefault, k8scorev1.DNSClusterFirstWithHostNet, k8scorev1.DNSClusterFirst, k8scorev1.DNSNone:
 			break
 		default:
 			allErrs = append(allErrs, field.Invalid(dnsPolicyPath, config.DNSPolicy, "invalid dns policy"))
 		}
 
-		if config.DNSPolicy == corev1.DNSNone {
+		if config.DNSPolicy == k8scorev1.DNSNone {
 			if config.DNSConfig != nil {
 				if len(config.DNSConfig.Nameservers) == 0 {
 					allErrs = append(allErrs, field.Required(dnsConfigPath.Child("nameservers"),
-						fmt.Sprintf("cannot be empty when dns policy is %s", corev1.DNSNone)))
+						fmt.Sprintf("cannot be empty when dns policy is %s", k8scorev1.DNSNone)))
 				}
 			} else {
 				allErrs = append(allErrs, field.Required(dnsConfigPath,
-					fmt.Sprintf("cannot be empty when dns policy is %s", corev1.DNSNone)))
+					fmt.Sprintf("cannot be empty when dns policy is %s", k8scorev1.DNSNone)))
 			}
 		}
 	}
 
+	if config.Devices != nil {
```

Comment:
The logic added starting with this line doesn't properly take into account the root volume, since it's not part of the `dataVolumes` slice passed. It needs to be passed independently via a new parameter to this function, e.g. `volume gardencorev1.Volume` and properly considered in the checks below.

Reply:
https://github.com/gardener/gardener-extension-provider-kubevirt/pull/87/files#r505654157